src/main/java/com/_up/megastore/data/repositories/ICategoryRepository.java

```diff
@@ -3,6 +3,9 @@
 import com._up.megastore.data.model.Category;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.Optional;
 import java.util.UUID;
 
-public interface ICategoryRepository extends JpaRepository<Category, UUID> {}
\ No newline at end of file
+public interface ICategoryRepository extends JpaRepository<Category, UUID> {
+    Optional<Category> findByCategoryIdAndDeletedIsFalse(UUID categoryId);
```

Comment:
Los optionals están buenos para cuando queremos obtener una entidad que después vamos a reutilizar o actualizar. En este caso, como sólo querés hacer la validación podrías refactorizar este método para que te devuelva un booleano directamente
```suggestion
    boolean existsByCategoryIdAndDeletedIsFalse(UUID categoryId);
```
De todos modos creo que este método podría no utilizarse

Reply:
pero no debía utilizar un JPA query method?